google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -273,14 +278,18 @@ private StreamWriter(Builder builder) throws IOException {
         }
       }
       this.location = location;
+      CredentialsProvider credentialsProvider = client.getSettings().getCredentialsProvider();
+      log.warning("The credentials gotten is " + credentialsProvider.getCredentials());
       // Assume the connection in the same pool share the same client and trace id.
       // The first StreamWriter for a new stub will create the pool for the other
       // streams in the same region, meaning the per StreamWriter settings are no
       // longer working unless all streams share the same set of settings
       this.singleConnectionOrConnectionPool =
           SingleConnectionOrConnectionPool.ofConnectionPool(
               connectionPoolMap.computeIfAbsent(
-                  ConnectionPoolKey.create(location),
+                  ConnectionPoolKey.create(
+                      location,
+                      credentialsProvider != null ? credentialsProvider.getCredentials() : null),
```

Comment:
Should we use a hash for the key instead? I am wondering if there could be a security concern here.

Reply:
Changed to use hash